make protobuf fails

as reported by @yifan-gu 

```
$ make protobuf
expr: syntax error
Invalidating /home/yifan/gopher/src/github.com/coreos/rkt/build-rkt-1.10.0+git/target/bin/rkt /home/yifan/gopher/src/github.com/coreos/rkt/build-rkt-1.10.0+git/stamps/__rkt_rkt_mk_home_yifan_gopher_src_github_com_coreos_rkt_build_rkt_1_10_0_git_target_bin_rkt_bgb_kv_depmk.stamp
Differing key: BGB_GO_FLAGS
Value stored in deps file: -ldflags " -X 'main.buildDefaultStage1Name=coreos.com/rkt/stage1-coreos' -X 'main.buildDefaultStage1Version=1.10.0+git5ec32ba' -X 'main.buildDefaultStage1ImageLoc=' -X 'github.com/coreos/rkt/version.Version=1.10.0+git5ec32ba' -X 'main.rktFeatures=+TPM +PROFILE' -X 'main.buildDefaultStage1ImageInRktDir=stage1-coreos.aci' -X 'main.buildDefaultStage1ImagesDir=/usr/local/lib/rkt/stage1-images' " -tags "selinux tpm profile"
Value currently used: -ldflags " -X 'main.buildDefaultStage1Name=coreos.com/rkt/stage1-coreos' -X 'main.buildDefaultStage1Version=1.10.0+git5ec32ba' -X 'main.buildDefaultStage1ImageLoc=' -X 'github.com/coreos/rkt/version.Version=1.10.0+git5ec32ba' -X 'main.rktFeatures=+TPM +PROFILE' -X 'main.buildDefaultStage1ImageInRktDir=stage1-coreos.aci' -X 'main.buildDefaultStage1ImagesDir=/usr/local/lib/rkt/stage1-images' " -tags "selinux tpm profile"
scripts/genproto.sh
building protoc-gen-go
can't load package: package vendor/github.com/golang/protobuf/protoc-gen-go: cannot find package "vendor/github.com/golang/protobuf/protoc-gen-go" in any of:
    /usr/local/go/src/vendor/github.com/golang/protobuf/protoc-gen-go (from $GOROOT)
    /home/yifan/gopher/src/github.com/coreos/rkt/build-rkt-1.10.0+git/gopath/src/vendor/github.com/golang/protobuf/protoc-gen-go (from $GOPATH)
rkt/rkt.mk:47: recipe for target 'protobuf' failed
make: *** [protobuf] Error 1
```

--- a/rkt/rkt.mk
+++ b/rkt/rkt.mk
@@ -43,8 +43,8 @@ bash-completion: | $(GOPATH_TO_CREATE)/src/$(REPO_PATH)
 		grep -vE '_test.go|main.go|_gen.go' | \
 		$(GO_ENV) xargs "$(GO)" run $(GOPATH_TO_CREATE)/src/$(REPO_PATH)/rkt/bash_completion_gen.go
 
-protobuf:
-	scripts/genproto.sh
+protobuf: | $(GOPATH_TO_CREATE)/src/$(REPO_PATH)
+	GOPATH=$(GOPATH_TO_CREATE) scripts/genproto.sh
 
 $(call undefine-namespaces,LOCAL)
 # RKT_STAMP deliberately not cleared

--- a/scripts/genproto.sh
+++ b/scripts/genproto.sh
@@ -16,9 +16,14 @@ if ! [[ $(protoc --version) =~ "3.0.0" ]]; then
 fi
 
 export PATH=.:${PATH}
+cd $GOPATH/src/github.com/coreos/rkt
 
 echo "building protoc-gen-go"
-go build vendor/github.com/golang/protobuf/protoc-gen-go
+pushd vendor/github.com/golang/protobuf/protoc-gen-go
+go build
+mv protoc-gen-go $(dirs -l +1)
+popd
+
 trap 'rm -f "protoc-gen-go"' EXIT
 
 echo "generating code"

